pkg/pattern/drain/drain.go

```diff
@@ -146,7 +146,7 @@ func DefaultConfig() *Config {
 	}
 }
 
-func New(config *Config) *Drain {
+func New(config *Config, metrics *Metrics) *Drain {
```

Comment:
I'm not sure if this is the best way to pass these metrics around. There are many streams/instances of Drain, so I had to register the metrics in the ingester code and pass it all the way through. Let me know if there's a preferred way to do this!

Reply:
yep this is it.